### PR TITLE
feat(developer-mining) replace BigInt with web3.js bignumber & refine affiliates calculation

### PR DESCRIPTION
**Pull request by chrismaree**
**Motivation**

Before this PR the affiliates calculation was still WIP and not fully functional. Also, bigInt was used in a few places. This PR finishes the affiliates calculation with associated tests and swaps out bigInt for bignumber.

**Summary**

Refine Affiliates calculation and replace all BigInt logic with bignumber.

**Details**

The internal logic of the affiliates script was someone redone to accommodate the required functionality.

**Issue(s)**

none.

### CODE REVIEW

File: packages/affiliates/test/affiliates.js [hunk 1]
```diff
@@ -26,8 +27,11 @@ function Queries() {
     streamBlocks() {
       return highland(require(`./datasets/${datasetName}/blocks`));
     },
-    getBlocks() {
-      return require(`./datasets/${datasetName}/blocks`);
+    getBlocks(start, end) {
+      return require(`./datasets/${datasetName}/blocks`).filter(block => {
+        const blockTime = moment(block.timestamp.value).valueOf();
+        return blockTime >= start && blockTime <= end;
```
Comment (daywiss) [inline packages/affiliates/test/affiliates.js:33]: nice improvement

**Review by daywiss — APPROVED** [review]

File: packages/affiliates/libs/affiliates.js [hunk 2]
```diff
@@ -65,18 +68,16 @@ const DeployerRewards = ({ queries, empCreatorAbi, empAbi, coingecko }) => {
...
   }
 
-  // update this function to calculate a price.
-  // may want to instead get price based on collateral and collaterlization ratio
-  // returns a BigInt
-  function calculateValue(tokens, closestPrice, decimals = 18) {
-    return BigInt(tokens) * BigInt(parseFixed(closestPrice.toFixed(decimals), decimals).toString());
+  // TODO: update this function to use underlying contracts GCR and total token debt.
+  function calculateValue(tokens, closestPrice, decimals) {
+    return tokens.mul(toBN(toWei(closestPrice.toString()))).div(toBN(toWei(decimals.toString())));
```
Comment (mrice32) [inline packages/affiliates/libs/affiliates.js:73]: just curious -- why prefer BN over BigInt?

File: packages/affiliates/libs/affiliates.js [hunk 3]
```diff
@@ -91,52 +92,60 @@ const DeployerRewards = ({ queries, empCreatorAbi, empAbi, coingecko }) => {
...
-        const contribution = totalValueLocked > 0n ? (BigInt(value) * (2n * 10n ** 18n)) / totalValueLocked : 0n;
-        const rewards = contribution * payoutPerSnapshot;
-        if (result[deployer] == null) result[deployer] = 0n;
-        result[deployer] = result[deployer] + rewards;
+        const contribution =
+          totalValueLocked.toString() != "0"
+            ? toBN(value) // eslint-disable-line indent
+                .mul(toBN(toWei("1"))) // eslint-disable-line indent
+                .div(totalValueLocked) // eslint-disable-line indent
+            : toBN("0"); // eslint-disable-line indent
```
Comment (mrice32) [inline packages/affiliates/libs/affiliates.js:134]: just curious -- why do we need all these eslint disables?

**Review by mrice32 — APPROVED** [review]
Looks great -- two minor questions

File: packages/affiliates/libs/affiliates.js (hunk 2 above)
Comment (chrismaree) [reply packages/affiliates/libs/affiliates.js:73]: the rest of the repo uses BigNumber & No where else do we use `BigInt`. Also, some numeric operations seemed to have less precision with `BigInt` (this was not verified just what my initial testing showed).

File: packages/affiliates/libs/affiliates.js (hunk 3 above)
Comment (chrismaree) [reply packages/affiliates/libs/affiliates.js:134]: prettier formats it like this for long length ternary operators. eslint doesn't like it!

**Pull request merged**